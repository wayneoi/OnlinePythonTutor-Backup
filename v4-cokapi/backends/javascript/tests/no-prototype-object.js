// 2020-02-10: almost works properly now, but there's a mysterious
// "TypeError: value.valueOf is not a function" error when running
// console.log() ... weird! anyhow, punt for now.
var y = Object.create(null);
console.log(y);
