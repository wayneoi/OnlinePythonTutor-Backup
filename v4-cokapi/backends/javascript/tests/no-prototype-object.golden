{ code: '// 2020-02-10: almost works properly now, but there\'s a mysterious\n// "TypeError: value.valueOf is not a function" error when running\n// console.log() ... weird! anyhow, punt for now.\nvar y = Object.create(null);\nconsole.log(y);',
  trace: 
   [ { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: { y: [ 'JS_SPECIAL_VAL', 'undefined' ] },
       ordered_globals: [ 'y' ],
       line: 4,
       col: 0,
       event: 'step_line',
       heap: {} },
     { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: { y: [ 'REF', 1 ] },
       ordered_globals: [ 'y' ],
       line: 5,
       col: 0,
       event: 'step_line',
       heap: { '1': [ 'INSTANCE', '' ] } },
     { exception_msg: 'TypeError: value.valueOf is not a function',
       stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: { y: [ 'REF', 1 ] },
       ordered_globals: [ 'y' ],
       line: 5,
       col: 0,
       event: 'exception',
       heap: { '1': [ 'INSTANCE', '' ] } },
     { stdout: '{}\n',
       func_name: '',
       stack_to_render: [],
       globals: { y: [ 'REF', 1 ] },
       ordered_globals: [ 'y' ],
       line: 5,
       col: 15,
       event: 'step_line',
       heap: { '1': [ 'INSTANCE', '' ] } } ] }
